src/Evolution/Systems/ScalarWave/BoundaryCorrections/UpwindPenalty.hpp

```diff
@@ -0,0 +1,288 @@
+ * \f}
+ *
+ * We denote the evolved fields by \f$u_{\alpha}\f$, the characteristic fields
+ * by \f$v_{\hat{\alpha}}\f$, and implicitly sum over reapeated indices.
+ * \f$T_{\alpha\hat{\alpha}}\f$ transforms characteristic fields to evolved
+ * fields, while \f$\Lambda_{\hat{\alpha}\hat{\beta}}^-\f$ is a diagonal matrix
+ * with only the negative characteristic speeds, and has zeros on the diagonal
+ * for all other entries. The int and ext superscripts denote quantities on the
+ * internal and external side of the mortar. Note that Eq. (6.3) of
+ * \cite Teukolsky2015ega is not exactly what's implemented since that boundary
+ * term does not consistently treat both sides of the interface on the same
+ * footing.
+ *
+ * For the scalar wave system the correction is:
+ *
+ * \f{align}{
+ *   D_{\Psi} &= \tilde{\lambda}_{v^{\Psi}}^{\mathrm{ext}}
+ *                v^{\mathrm{ext},\Psi}
+ *                - \tilde{\lambda}_{v^{\Psi}}^{\mathrm{int}}
+ *                v^{\mathrm{int},\Psi}, \\
+ *   D_{\Pi} &= \frac{1}{2}\left(\tilde{\lambda}_{v^+}^{\mathrm{ext}}
+ *             v^{\mathrm{ext},+} +
+ *             \tilde{\lambda}_{v^-}^{\mathrm{ext}}
+ *             v^{\mathrm{ext},-}\right)
+ *             + \tilde{\lambda}_{v^\Psi}^\mathrm{ext}\gamma_2
+ *             v^{\mathrm{ext},\Psi}
+ *             \notag \\
+ *           &-\frac{1}{2}\left(\tilde{\lambda}_{v^+}^{\mathrm{int}}
+ *             v^{\mathrm{int},+} +
+ *             \tilde{\lambda}_{v^-}^{\mathrm{int}}
+ *             v^{\mathrm{int},-}\right)
+ *             - \tilde{\lambda}_{v^\Psi}^\mathrm{int}\gamma_2
+ *             v^{\mathrm{int},\Psi} , \\
+ *   D_{\Phi_{i}}
+ *              &= \frac{1}{2}\left(\tilde{\lambda}_{v^+}^{\mathrm{ext}}
+ *                v^{\mathrm{ext},+}
+ *                - \tilde{\lambda}_{v^-}^{\mathrm{ext}}
+ *                v^{\mathrm{ext},-}\right)n_i^{\mathrm{ext}}
+ *                + \tilde{\lambda}_{v^0}^{\mathrm{ext}}
+ *                v^{\mathrm{ext},0}_{i}
+ *                \notag \\
+ *              &-
+ *                \frac{1}{2}\left(\tilde{\lambda}_{v^+}^{\mathrm{int}}
+ *                v^{\mathrm{int},+}
+ *                - \tilde{\lambda}_{v^-}^{\mathrm{int}}
+ *                v^{\mathrm{int},-}\right)n_i^{\mathrm{int}}
+ *                - \tilde{\lambda}_{v^0}^{\mathrm{int}}
+ *                v^{\mathrm{int},0}_{i},
+ * \f}
+ *
+ * with characteristic fields
+ *
+ * \f{align}{
+ *   v^{\Psi} &= \Psi, \\
+ *   v^{0}_{i} &= (\delta^k_i-n^k n_i)\Phi_{k}, \\
+ *   v^{\pm} &= \Pi\pm n^i\Phi_{i} -\gamma_2 \Psi,
+ * \f}
+ *
+ * and characteristic speeds
+ *
+ * \f{align}{
+ *   \lambda_{v^\Psi} =& -v^i_g n_i, \\
+ *   \lambda_{v^0} =& -v^i_g n_i, \\
+ *   \lambda_{v^\pm} =& \pm 1 - v^i_g n_i,
+ * \f}
+ *
+ * where \f$v_g^i\f$ is the mesh velocity and \f$n_i\f$ is the outward directed
+ * unit normal covector to the interface. We have also defined
+ *
+ * \f{align}{
+ *   \tilde{\lambda}_{\hat{\alpha}} =
```

Comment:
In lines 40-50ish, you use `\Lambda^{-}` for a matrix with only negative char speeds. It would be notationally simpler to use `\lambda^{-}` (lower case lambda) for each of these char speeds, instead of introducing a new symbol (tilde) to denote the same idea.